tests/structure.test.ts

```diff
@@ -317,4 +432,18 @@ describe('Internal tree structure', () => {
     }
     expect(result_iterate_data).toEqual(result_for_loop);
   });
+
+  it('should find index based on a given condition', () => {
```

Comment:
I don't know whether this is the expected behavior (this method is not used anywhere yet) - maybe we can improve the documentation @spion ?

Reply:
We'll probably be exposing these (similarly to how reduceTo is exposed) when we need to implement monoids.